graphql_compiler/schema_merging/rename_schema.py

```diff
@@ -0,0 +1,232 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+from collections import namedtuple
+
+from graphql import build_ast_schema, parse
+from graphql.language.visitor import Visitor, visit
+
+from .utils import SchemaError, get_schema_data
+
+
+RenamedSchema = namedtuple(
+    'RenamedSchema', [
+        'schema_ast',  # type: Document, ast representing the renamed schema
+        'reverse_name_map',  # type: Dict[str, str], new type name to old type name
+        'reverse_root_field_map'  # type: Dict[str, str], new root field name to old root field
+    ]
+)
+
+
+def rename_schema(schema_string, rename_func=lambda name: name):
+    """Create a RenamedSchema, where types and query root fields are renamed using rename_func.
+
+    Args:
+        schema_string: string describing a valid schema that does not contain extensions
+        rename_func: callable that takes string to string, used to transform the names of
+                     types, interfaces, enums, and root fields. Defaults to identity
+
+    Returns:
+        RenamedSchema
+
+    Raises:
+        SchemaError if input schema_string does not represent a valid schema without extensions,
+            or if there are conflicts between the renamed types or root fields
+    """
+    # Check that the input string is a parseable and valid schema.
+    try:
+        ast = parse(schema_string)
+        build_ast_schema(ast)
+    except Exception as e:  # Can't be more specific -- see graphql/utils/build_ast_schema.py
+        raise SchemaError('Input schema does not define a valid schema.\n'
```

Comment:
Two minor nits: Let's use an `u'foo'` explicit unicode string instead of `'foo'`, for the sake of Python 2. And let's avoid newlines in error messages, since sometimes they fail to get correctly represented through logs and other mechanisms.